testing-modules/testing-libraries-2/src/test/java/com/baeldung/environmentvariablesfortest/EnvironmentVariablesSystemRulesUnitTest.java

```diff
@@ -0,0 +1,23 @@
+package com.baeldung.environmentvariablesfortest;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.contrib.java.lang.system.EnvironmentVariables;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class EnvironmentVariablesSystemRulesUnitTest {
+    @Rule
+    public EnvironmentVariables environmentVariablesRule = new EnvironmentVariables();
+
+    @Before
+    public void before() {
+        environmentVariablesRule.set("system rules", "works");
```

Comment:
You can set an environment variable at any point. During before, `System.getenv` won't return the value you've just set, because env substitution isn't active. Within the test, the `set` method is "live"